api/addons/v1beta1/conversion_test.go

```diff
@@ -70,3 +71,21 @@ func spokeClusterResourceSetStatus(in *ClusterResourceSetStatus, c randfill.Cont
 		}
 	}
 }
+
+func ClusterResourceSetBindingFuzzFuncs(_ runtimeserializer.CodecFactory) []interface{} {
+	return []interface{}{
+		hubClusterResourceSetStatus,
+		spokeClusterResourceSetBindingSpec,
+	}
+}
+
+func spokeClusterResourceSetBindingSpec(in *ClusterResourceSetBindingSpec, c randfill.Continue) {
+	c.FillNoCustom(in)
+
+	for i, b := range in.Bindings {
+		if b != nil && reflect.DeepEqual(*b, ResourceSetBinding{}) {
+			b = nil
+		}
+		in.Bindings[i] = b
```

Comment:
Is this capturing the loop variable?

Reply:
Not sure if I got the question right :)

They changed something in a Go version a while back that it's now safe to use the pointer of a loop variable, but this doesn't mean that when you modify the loop variable this also modifies the underlying array.

IIRC b is a copy of the array element.

In any case, without this line the binding in the underlying array is not modified

Reply:
Btw, this is a pretty unrealistic case anyway as ResourceSetBinding has a Required field with MinLength = 1.

I just fixed up that case enough so that conversion tests are not failing

Reply:
Yeah I was just wanting to make sure that we aren't converting the list to a list of all the same thing, as that was a classic gotcha when you assigned a list in a loop like this.

Don't know if there's an easy way to quickly visual that this isn't the case now?

Since `b` is a pointer to a `ResourceBindingSet` right?

Reply:
I slightly optimized this code. What do you think about the new version?

Reply:
Clearer to me yep